jetty-ee10/jetty-ee10-plus/src/main/java/org/eclipse/jetty/ee10/plus/webapp/EnvConfiguration.java

```diff
@@ -241,6 +230,43 @@ protected void createEnvContext(WebAppContext wac)
         }
     }
 
+     /**
+     * Obtain a WEB-INF/jetty-ee10-env.xml, falling back to
+     * looking for WEB-INF/jetty-env.xml.
+     *
+     * @param webInf the WEB-INF of the context to search
+     * @return the file if it exists or null otherwise
+     */
+    private Resource resolveJettyEnvXml(Resource webInf)
+    {
+        String xmlFile = JETTY_EE10_ENV_XML_FILENAME;
+
+        try
+        {
+            if (webInf == null || !webInf.isDirectory())
+                return null;
+
+            //try to find jetty-ee10-env.xml
+            Resource xmlResource = webInf.resolve(xmlFile);
+            if (!Resources.missing(xmlResource))
+                return xmlResource;
+
+            //failing that, look for jetty-env.xml
+            xmlFile = JETTY_ENV_XML_FILENAME;
+            xmlResource = webInf.resolve(xmlFile);
+            if (!Resources.missing(xmlResource))
+                return xmlResource;
+
+            return null;
+        }
+        catch (Exception e)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Error resolving WEB-INF/" + xmlFile, e);
+            return null;
+        }
+    }
```

Comment:
I think the following is clearer and the exception will have the detail needed:

```suggestion
    private Resource resolveJettyEnvXml(Resource webInf)
    {
        try
        {
            if (webInf == null || !webInf.isDirectory())
                return null;

            //try to find jetty-ee10-env.xml
            Resource xmlResource = webInf.resolve(JETTY_EE10_ENV_XML_FILENAME);
            if (!Resources.missing(xmlResource))
                return xmlResource;

            //failing that, look for jetty-env.xml
            xmlResource = webInf.resolve(JETTY_ENV_XML_FILENAME);
            if (!Resources.missing(xmlResource))
                return xmlResource;

            return null;
        }
        catch (Exception e)
        {
            if (LOG.isDebugEnabled())
                LOG.debug("Error resolving", e);
            return null;
        }
    }
```